scene/main/canvas_item.h

```diff
@@ -237,7 +237,7 @@ class CanvasItem : public Node {
 	Color get_modulate() const;
 	Color get_modulate_in_tree() const;
 
-	void set_self_modulate(const Color &p_self_modulate);
+	virtual void set_self_modulate(const Color &p_self_modulate);
```

Comment:
Seems like it supersedes #75320? (and likely fixes the same issue)

Reply:
The point of making this virtual is for the TileMapLayer to trigger an update. It is needed because the modulation for the selected layer depends on this (basically, it overrides self_modulate on the layer canvas item. So if you modify this value, we need to darken it before applying it again).
 
It does not fix the issue, but I am not sure it is worth fixing TBH. The way to solve it would be to multiply the self-modulate value of the layer by the parent TileMap's one, but only if it is an internal node. That's kind of additional glue code that will not be required anyway with the new workflow. So, well, not sure how much we want to keep the deprecated workflow alive. 

In any case, the whole point of the PR is to make TileMapLayer less dependent on TileMap for now, so I'd rather fix that later on.

Reply:
I mean TileMapLayer `self_modulate` has no effect. This at least should be fixed.
Also if it's only used for darkening then it's pointless right now, because you can't change `self_modulate` inside editor without using hacks to make the node appear in scene.

Reply:
> I mean TileMapLayer self_modulate has no effect. This at least should be fixed.

Well, I don't know. Something like the ColorPicker node, which has internal sub-nodes, has a self_modulate property that does nothing too. Solving it kind of requires the implementing something like `TileMapLayer::set_layer_as_internal(...)` so that the TileMapLayer behaves differently depending on whether it is an internal node of the TileMap or not. 

I feel it would be a bit hacky, even if I agree it seems like more practical to have self-modulate doing something. So well, I guess it's probably worth it.

>  Also if it's only used for darkening then it's pointless right now, because you can't change self_modulate inside editor without using hacks to make the node appear in scene. (also DIRTY_FLAGS_LAYER_MODULATE looks unused)

Ah right. but it will be useful in the new workflow. Let's say it's future-proof then. :grin:

Reply:
Why being internal matters here? When layer will be visible in the editor, users will expect that it's `self_modulate` is functional.
And again, this has nothing to do with the parent TileMap node.

Reply:
I've just pushed an update, modifying `TileMapLayer`'s self_modulate in the remote tree should work now.